### ISSUE TITLE
It seems like your access key is not valid

### ISSUE THREAD

**Issue by hadifarnoud** [Jigsaw-Code/outline-server#210]
I had a VPS with only outline on it. it was working fine until I ran into `It seems like your access key is not valid` issue.

I rebuilt my server (fresh ubuntu 16.04 install then Outline installed). after 3 days I got `It seems like your access key is not valid` error again. 

I asked a friend of mine in US to test the connection and she said it's working fine. There seems to be a new method of blocking Outline in Iran. I'm going to keep this server as is for testing. Let me know how we can find out what's happening

**Comment by hadifarnoud** [comment]
it's simply unusable as it is. gets blocked so quickly. I did manage to use my server for one day only after not connecting for a few days. same error again now.

**Comment by hadifarnoud** [comment]
update: it's totally random. sometimes it's working for a few days with no issue. sometimes it gets reblocked quickly.

**Comment by fortuna** [comment]
That's frustrating. :-(
We are theorizing that some locations are blocked more than others. Where are you running your servers? Have you tried different locations? Do they get blocked as quickly?
Thanks for the update!

**Comment by hadifarnoud** [comment]
I did figure that out some time ago. Looks like Digitalocean is specially worse. Good news is, they don't block your server forever. It looks like they are not sure server is for proxy. I'm guessing here

**Comment by mehranhassani** [comment]
For me, US, India, and Singapore servers on Digital Ocean are usually blocked within a week. My last server was Toronto and its working fine for a while now, but I think its completely random as @hadifarnoud mentioned. My AWS didn't get blocked for a month but I switched to DO after so I am not sure. The one on AWS was an east US.

**Comment by Nocturrne** [comment]
I retested this today, just to confirm what I already know. China and their Great Firewall have been able to automatically detect and block Shadowsocks for a long time. The concept of wrapping a VPN client and server into a nice UI is very good, but you'll need much much more than this to accomplish your goal. Seriously, am I very disappointed with Google/Alphabet - you have the resources and ability to change the internet, but you won't do it because privacy would break your business model. Eric Schmidt, Larry Page, Sergey Brin - you should be ashamed.

**Comment by fortuna** [comment]
@Nocturrne , could you give us details on the methodology of your test? What did you do, and what did you find?

To build a hard-to-block VPN solution is a huge challenge, and having input from the community is very valuable.

**Comment by hadifarnoud** [comment]
@fortuna I appreciate the effort. It's kinda hard to build a censorship buster from far. It's a cat and mouse game in a sense.

I can confirm that shadowsocks does not work well in Iran either. probably Iran has the same tech

**Comment by fortuna** [comment]
@hadifarnoud Next time you have trouble accessing your server, would you minding testing the acces with this tool? https://fortuna-outline-connectivity.netlify.com/

It can give us more information on whether the server and port are blocked, and how that's done.

**Comment by Nocturrne** [comment]
I did some tests in multiple cities in China, using several different mobile devices and PCs - the results were the same everywhere.  The great firewall detects shadowsocks immediately and packets get dropped.  On PC, the client will respond with "access key is not valid" and on Android the client will appear to connect, but there is no routing.  As soon as I stepped out of China, clients on all of my devices were able to connect without any problems.

Server Connectivity Test:
Your network does not block port 443
Host -.-.-.- is blocked by packet dropping or server is down

I have had the same results with my personal shadowsocks servers over the last year, so shut them down.  An open source solution is just too easy for the state to analyze.

**Comment by fortuna** [comment]
@Nocturrne thanks for the info. Unfortunately the test was not meaningful for ports < 1024 for various reasons, so we can't fully trust those results.

I've updated the test to consider those ports. However, it still assumes that there's nothing running on your port 80 (we need to use as a baseline a port that we know is closed and the network doesn't block, so we assume 443 or 80).

But it I believe they are blocking your server. I wonder if they check `is this traffic to port 443 using TLS?`. That protocol check is easy to do. Have you tried other port numbers?

Are you able to access other services on your server, such as ssh or a webserver?
Are you able to ping the server?

One fun test you can do is TCP traceroute. It can tell you exactly where the blocking is happening:
```
sudo traceroute --tcp -O info -p $PORT $HOST
```

**Comment by rouzwelt** [comment]
so getting this error too, I just installed outline server in my ops (vultr), did everything by the book, but I cannot connect with client app, and getting that error, I have no issue connecting to my vps and ssh into it, but it is just that client app cant get connected,
it is worth mentioning that Im in Iran, I asked one of my friends in south america to test and he was wable to get connected and everything fine for him, but not for me, any workaround for this?

**Comment by shahabedinh** [comment]
I created some keys and shared with some friends. In two devices, I see the "access key is not valid"  error. I created new keys even in another server, but does not solve the issue

**Comment by rouzwelt** [comment]
@shahabedinh yea exactly same issue on my side as well, I guess it is Iran's DPI figuring it out and dropping the connection, but I know some of my fiends are already using Outline, they use the public keys posted in the Outline's Telegram channel daily and they are able to connect, so clearly there should be some config/workaround somewhere on the server that can get this to work.
I'd really appreciate it if Outline team can help us with it. 

Need to mention again that I can ssh into my server with no issue without any VPNs, so the server is not blocked itself, but using the Outline protocol (Shadowsocks) is what that results in this error and not being able to connect.

**Comment by Follow-The-Rabbit** [comment]
Hey @fortuna, did you find out more about the error with validateServerCredentials () ?
